plugin/src/com/microsoft/alm/plugin/idea/tfvc/core/tfs/TFVCUtil.java

```diff
@@ -97,10 +98,12 @@ public static boolean isInvalidTFVCPath(@NotNull TFSVcs vcs, @NotNull FilePath p
         return filteredPaths;
     }
 
-    private static boolean isInServiceDirectory(FilePath filePath) {
-        String path = filePath.getPath();
-        return StringUtils.containsIgnoreCase(path, "$tf")
-                || StringUtils.containsIgnoreCase(path, ".tf");
+    /**
+     * Determines whether the file is in the TFVC service directory ($tf or .tf).
+     */
+    public static boolean isInServiceDirectory(String filePath) {
```

Comment:
I've been trying to reduce amount of memory traffic in performance-sensitive places. If we're to create a `new FilePath` every time user changes a file, then the memory could go crazy very fast on big TFVC repositories. That's why I've changed _some_ of the methods called most often during the `tf status` operation to accept plain `String`s instead of `String`s wrapped into any heap-allocated wrappers.

Reply:
In all usages of that method, you call path.getPath() I do not understand how it reduces traffic.

Reply:
After closer inspection, I agree, it doesn't. Will roll it back.